### PR TITLE
Reorganize workflows

### PR DESCRIPTION
**Pull request by lukasmalkmus**
This reorganizes the single workflow into multiple ones. It also gets rid of a lot of blank lines, self explaining descriptions, etc. Also improves caching performance.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: PR
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (bahlo) [inline .github/workflows/pr.yml:6]: Can we run this workflow  on master as well to catch semantic merge conflicts? And maybe releases as well just to be safe.

**Review by bahlo — CHANGES_REQUESTED** [review]
Small question

File: .github/workflows/pr.yml (hunk 1 above)
Comment (lukasmalkmus) [reply .github/workflows/pr.yml:6]: We could. But the repo settings just allow rebase commits so the HEAD of your PR branch represents the HEAD of the main branch, if the PR lands.

Comment (bahlo) [reply .github/workflows/pr.yml:6]: TBH even with the checks I'd feel way better if the tests ran on `main` and tags.

Comment (lukasmalkmus) [reply .github/workflows/pr.yml:6]: Well, I'd say, only `main`. Because when we tag a release, we do that on a main branch that passed already, right?

**Review by bahlo — APPROVED** [review]

**Pull request merged**